openedx/frontend-app-course-authoring
# Taxonomy list page [FC-0036]

**Changes added:**

- Add `tsconfig.json` to configure TypeScript to run ts-check on taxonomy files
- Build the [Taxonomy List page](https://github.com/openedx/modular-learning/issues/104)

**Screenshots**

![image](https://github.com/openedx/frontend-app-course-authoring/assets/11827305/6a1ae462-6a55-4bad-ab93-4d39cebfa06e)


**Testing instructions**

- Follow the testing instructions of the [edx-platform PR](https://github.com/openedx/edx-platform/pull/33439) to setup your devstack.
- You can to the [Taxonomy List page here](http://localhost:2001/taxonomy-list)
- Verify that system-defined taxonomies have been loaded.
- Go to the python CMS shell and create a new Taxonomy:

```
from openedx.core.djangoapps.content_tagging.api import create_taxonomy
create_taxonomy(name='Taxonomy', description='This is a Description')
```

- Check the new taxonomy on the list.
- Creata a taxonomy with a large name and a large description.
- Check the new taxonomy on the list.
- Create a disabled taxonomy:

```
create_taxonomy(name='Taxonomy 2', description='This is a Description', enabled=False)
```

- Check that the new taxonomy doesn't appear on the list.
- Edit the description of a system-defined taxonomy with a long description:

```
from openedx_tagging.core.tagging.models import Taxonomy
lang = Taxonomy.objects.get(pk=-1)
lang.description = 'Put a long description here....'
lang.save()
```

-  Check the card change on the list.

**Other info**

- Internal ticket: [FAL-3522](https://tasks.opencraft.com/browse/FAL-3522)
- edx-platform realted PR: https://github.com/openedx/edx-platform/pull/33439
- openedx-learning realted PR: https://github.com/openedx/openedx-learning/pull/98
- Closes https://github.com/openedx/modular-learning/issues/104
